### DISCUSSION TITLE
Illegal operations at \test\correctness\multi_splits_with_diff_tail_strategies.cpp

### DISCUSSION THREAD

**Discussion by shaniaba** [halide/Halide#7939] [Q&A]
Hello

I think there are 2 illegal operations at the test \test\correctness\multi_splits_with_diff_tail_strategies.cpp when tail_strategy equals TailStrategy::PredicateLoads -
1. the split operation must be done on the innermost loop - which is "c" (after reorder) and not "y". 
2. the output size must to be a multiple of the split factor - which 64 is not a multiple of 2000.
Those facts are written at Schedule.h regard TailStrategy::PredicateLoads.

What do you think? Am I right?

Thanks a lot

**Comment by abadams** [comment]
PredicateLoads must be the outermost split *in that dimension*. In this case it's the outermost split that originates from the Var 'y', so it's fine. The comment needs fixing.

The output for that Func must indeed be a multiple of the split factor, and this would throw an error in AOT code, but the JIT is capable of querying this of the compiled pipeline, realizing a slightly larger buffer, and returning a cropped view of it.
